hazelcast/src/main/java/com/hazelcast/jet/impl/util/ExceptionUtil.java

```diff
@@ -235,6 +235,9 @@ public static boolean isTechnicalCancellationException(Throwable t) {
         Throwable peeledFailure = peel(t);
         return peeledFailure instanceof JobTerminateRequestedException
                 || peeledFailure instanceof ResultLimitReachedException
-                || peeledFailure instanceof TerminatedWithSnapshotException;
+                || peeledFailure instanceof TerminatedWithSnapshotException
+                ||
```

Comment:
```java
|| peeledFailure instanceof ResultLimitReachedException
                || peeledFailure instanceof TerminatedWithSnapshotException;
                || peeledFailure instanceof TerminatedWithSnapshotException
```

you can change to:
```java
isTechnicalCancellationException(peeledFailure) || (peeledFailure != peeledFailure.getCause()
                        && isTechnicalCancellationException(peeledFailure.getCause()))
```

Reply:
Needed to add a null-check, see the update.